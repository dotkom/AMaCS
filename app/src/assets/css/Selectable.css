
.container {
  background: #ffffff;
  border: 2px solid #fff;
  padding: 0.6em 0.75em;
  box-shadow: 0 3px 4px rgba(0, 0, 0, 0.2);
  display: inline-flex;
  max-height: 100%;
  width: 100%;
  max-width: 5.5rem;
  align-items: center;
  flex-direction: column;
  border-radius: 2px;
  transition: 0.25s border-color;
  user-select: none;
  cursor: default;
}

.container:hover, .container.container.selected:hover {
  border-color: #0060a3;
}

.container.small {
  font-size: 0.85rem;
}

.container.selected {
  background: #C4C4C4;
  border-color: #f2f2f2;
  box-shadow: inset 0px 3px 4px rgba(0, 0, 0, 0.2);
  border-radius: 3px;
}

.container svg {
  height: 100%;
  width: 100%;
}

.name {
  margin: 0;
  font-size: 0.9em;
  font-weight: 600;
}


@media (min-width: 600px) {
  .container svg {
    height: 4rem;
    width: 4rem;
  }
}
