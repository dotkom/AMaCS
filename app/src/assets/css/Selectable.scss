@import 'variables';

.container {
  background: white;
  border: 2px solid white;
  padding: 0.6em 0.75em;
  box-shadow: 0 3px 4px rgba(0, 0, 0, 0.2);
  display: inline-flex;
  max-height: 100%;
  width: 100%;
  max-width: 5.5rem;
  align-items: center;
  flex-direction: column;
  border-radius: 2px;
  transition: 0.25s border-color;
  user-select: none;
  cursor: default;
}

.container.draggable:hover, .container.draggable.selected:hover {
  border-color: $blue;
}

.container.small {
  font-size: 0.85rem;
}

.container.selected {
  background: #C4C4C4;
  border-color: $light-gray;
  box-shadow: inset 0px 3px 4px rgba(0, 0, 0, 0.2);
  border-radius: 3px;
}

.container svg {
  height: 100%;
  width: 100%;
}

.name {
  margin: 0;
  font-size: 0.9em;
  font-weight: 600;
}


@media (min-width: 600px) {
  .container svg {
    height: 4rem;
    width: 4rem;
  }
}
