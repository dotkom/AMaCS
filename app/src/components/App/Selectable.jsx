import React from "react";
import PropTypes from 'prop-types';

import _s from "assets/css/Selectable.css";
import classNames from "classnames";


function Selectable({ committee, onClick, small, selected }) {
  const Icon = committee.icon;
  return(
    <div className={classNames(_s.container, { [_s.small]: small, [_s.selected]: selected })} onClick={onClick}>
      <Icon width={64} height={64} />
      <h2 className={classNames(_s.name)}>{committee.name}</h2>
    </div>
  )
}

Selectable.defaultProps = {
  small: false,
  onClick: null,
  selected: false
}

Selectable.propTypes = {
  committee: PropTypes.shape({
    icon: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
    name: PropTypes.string,
  }).isRequired,
  onClick: PropTypes.func,
  small: PropTypes.bool,
  selected: PropTypes.bool
}

export default Selectable;
