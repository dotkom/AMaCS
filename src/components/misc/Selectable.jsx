import React from 'react';
import classNames from 'classnames';

import _s from 'assets/css/Selectable.module.scss';
import { API_SETTINGS } from 'common/constants';

const Selectable = ({ committee, onClick = null, small = false, selected = false }) => {
  return (
    <div className={classNames(_s.container, { [_s.small]: small, [_s.selected]: selected })} onClick={onClick}>
      <img className={_s.img} src={`${API_SETTINGS.host}${committee.image.md}`} alt={committee.name_long} />
      <h2 className={classNames(_s.name)}>{committee.name_short}</h2>
    </div>
  );
};

export default Selectable;
