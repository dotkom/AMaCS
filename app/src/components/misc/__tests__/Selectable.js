import React from 'react';
import { shallow } from 'enzyme';

import Selectable from '../Selectable';

describe('Selectable', () => {
  const committee = {
    name: 'Testkom',
    icon: 'icon-here',
  };
  
  it('renders correctly', () => {
    const wrapper = shallow(
      <Selectable committee={committee} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('calls onClick when clicked', () => {
    const onClickMock = jest.fn();
    const wrapper = shallow(
      <Selectable committee={committee} onClick={onClickMock} />
    );
    wrapper.simulate('click');
    expect(onClickMock).toHaveBeenCalled();
  });
});
