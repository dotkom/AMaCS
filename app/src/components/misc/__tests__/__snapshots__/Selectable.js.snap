// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selectable renders correctly 1`] = `
<div
  className="container"
  onClick={null}
>
  <icon-here
    height={64}
    width={64}
  />
  <h2
    className="name"
  >
    Testkom
  </h2>
</div>
`;
